core/imt/src/TThreadExecutor.cxx

```diff
@@ -183,6 +195,12 @@ double TThreadExecutor::ParallelReduce(const std::vector<double> &objs,
 float TThreadExecutor::ParallelReduce(const std::vector<float> &objs,
                                       const std::function<float(float a, float b)> &redfunc)
 {
+   if (GetPoolSize() > tbb::global_control::active_value(tbb::global_control::max_allowed_parallelism)) {
+      Warning("TThreadExecutor::ParallelReduce",
+              "tbb::global_control is limiting the number of parallel workers."
+              " Proceeding with %zu threads this time",
+              tbb::global_control::active_value(tbb::global_control::max_allowed_parallelism));
+   }
```

Comment:
this logic repeated 3 times could be factored out in a little helper function

Reply:
For these three times in such a short file, I feel creating a helper to print would be overkill, and creating a helper that includes the call to GetPoolSize() would complicate the code.